XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs

```diff
@@ -81,8 +81,8 @@ public List<XenServerVersion> XenServerVersionsForAutoCheck
         public DownloadUpdatesXmlAction(bool checkForXenCenter, bool checkForServerVersion, bool checkForPatches, string userAgent, string userAgentId, string checkForUpdatesUrl = null)
             : base(null, "_get_updates", "_get_updates", true)
         {
-            Debug.Assert(checkForUpdatesUrl != null, "Parameter checkForUpdatesUrl should not be null. This class does not default its value anymore.");
-            Debug.Assert(!string.IsNullOrWhiteSpace(userAgent) && !string.IsNullOrWhiteSpace(userAgentId));
+            //Debug.Assert(checkForUpdatesUrl != null, "Parameter checkForUpdatesUrl should not be null. This class does not default its value anymore.");

```

Comment:
We shouldn't modify this class in XenModel. Instead we should fix its usages in CFUValidator so that the addertion is not raised.